sklearn/decomposition/online_lda.py

```diff
@@ -719,3 +750,34 @@ def perplexity(self, X, doc_topic_distr=None, sub_sampling=False):
         perword_bound = bound / word_cnt
 
         return np.exp(-1.0 * perword_bound)
+
+    def perplexity(self, X, doc_topic_distr=None, sub_sampling=False):
+        """Calculate approximate perplexity for data X.
+
+        Perplexity is defined as exp(-1. * log-likelihood per word)
+
+        .. versionchanged:: 0.19
+           *doc_topic_distr* argument has been depricated because user no
+           longer has access to unnormalized distribution
+
+        Parameters
+        ----------
+        X : array-like or sparse matrix, [n_samples, n_features]
+            Document word matrix.
+
+        doc_topic_distr : None or array, shape=(n_samples, n_topics)
+            Document topic distribution.
+            If it is None, it will be generated by applying transform on X.
+
+        Returns
+        -------
+        score : float
+            Perplexity score.
+        """
+
+        if doc_topic_distr is not None:
+            warnings.warn("Argument 'doc_topic_distr' is deprecated and will "
```

Comment:
Technically this should also show if the user provides an explicit value for `doc_topic_distr`. So you need a sentinel other than `None`, such as `'deprecated'`